AY2122S1-CS2103T-T17-3/tp
# Fix tag panel bug

### Description

Fix the bugs with tag panel not updating properly.

### Type of change


### How to test

Things to test (non-exhaustive)

1. Delete client with unique tag: delete that unique tag from tag panel
2. Add client with new tag: add new tag to tag panel
3. Add client with existing tag: not add duplicate tag to tag panel
4. Switch addressbook: switch tag in tag panel to new addressbook tag
5. Switch between addressbook: tag panel should be updated accordingly
6. Edit client to add new tag: add new tag to tag panel
7. Edit client who have unique tag to new tag: unique tag to be removed and new tag to be added
8. Edit client who have unique tag to existing tag: unique tag to be removed and no new tag added
9. Edit existing -> new
10. Edit existing -> existing
12. Edit existing -> no tag
13. Edit no tag -> existing
14. Edit no tag -> new
15. Edit no tag -> no tag
16. Initialise new addressbook: tag should be displayed accordingly

### Checklist
